### PR TITLE
18MEX remove P2's ability if Copper Canyon is laid [Fixes #2855]

### LINKED ISSUE #2855: 18MEX - track step blocks even though there's no valid track action
https://18xx.games/game/12969?action=152

CHI owns KCMO, and KCMO hasn't used its ability, so this step blocks; however, KCMO cannot use its ability as Copper Canyon (F5) has already had a tile placed.

`Step::18Mex::Track#remaining_tile_lay?` should probably check for `copper_canyon_hex_empty?`, like `Step::18Mex::SpecialTrack#tile_lay_abilities` does, or even better, if we could genericize a tile-laying ability being removed when its corresponding hex is laid, then this could be handled in the config with an extra field on KCMO's ability.

(I am considering making a new issue for tile-laying abilities and a separate one for teleport abilities and throwing checklists in them, but didn't want to interrupt what I'm doing now to take the time to do that and didn't want to forget about this particular case, so just making another issue for now)

### PR DESCRIPTION
**Pull request by michaeljb**
I have a local migration for this that fixes 55/60 of the broken cases. The last 5 can't be fixed by that migration and are all finished anyway; they'll need more fixing anyway for the changes I have in progress for the [rest of the issues I'm working on](https://github.com/tobymao/18xx/issues?q=is%3Aopen+is%3Aissue+assignee%3Amichaeljb+label%3A%22ability+timing%22).

### CODE REVIEW

**Review by tobymao — APPROVED** [review]

**Comment by perwestling** [comment]
It should be noted the the KCM&O private should be closed when ability is used. That is according to the information from the designer:
https://boardgamegeek.com/thread/535347/error-kcmo-private-company

**Comment by perwestling** [comment]
@michaeljb This has also been clarified in the Wiki page (https://github.com/tobymao/18xx/wiki/18MEX#rules-clarifications) and fixed 2020-11-22 at 18xx.games.

**Comment by perwestling** [comment]
Mea culpa....

You are right. The ability should be removed if someone has laid a tile in the Copper Canyon hex. 
So yes, this is a fine commit.

File: lib/engine/game/g_18_mex/game.rb [hunk 1]
```diff
@@ -1090,6 +1090,17 @@ def train_limit(entity)
           super + Array(abilities(entity, :train_limit)).sum(&:increase)
         end
 
+        def action_processed(action)
+          case action
```
Comment (perwestling) [inline lib/engine/game/g_18_mex/game.rb:1094]: Should it not be super first, and then case?

**Review by perwestling — COMMENTED** [review]
I am just wondering about a detail.

**Review by tobymao — APPROVED** [review]

File: lib/engine/game/g_18_mex/game.rb (hunk 1 above)
Comment (michaeljb) [reply lib/engine/game/g_18_mex/game.rb:1094]: Hmm maybe it should. I was aping how 18Chesapeake does `action_processed`, which also doesn't call `super`. In `Game::Base`, `action_processed` just checks for corporations to close, which doesn't apply to Chesapeake or MEX, so it'll work without this, but we'll run into trouble if `Game::Base#action_processed` gets more functionality put in.

**Pull request merged**